web/src/api/api.ts

```diff
@@ -86,4 +86,41 @@ export class ImmichApi {
 	}
 }
 
+export type MapMarkerResponseDto = [
+	// latitude
+	number,
+	// longitude
+	number,
+	// assetId
+	string
+];
+
+export class MapMarkerResponseDtoDecorator {
```

Comment:
That's an unusual way of doing this conversion. What do you think about the following implementation?

```typescript
export type MapMarkerResponseDto = [
	// latitude
	number,
	// longitude
	number,
	// assetId
	string?
];

export type MapMarker = {
	assetId?: string;
	location: [number, number];
};

export function convertObjectToMarker(obj: object): MapMarker {
	const [lat, lon, assetId] = obj as MapMarkerResponseDto;

	return {
		assetId,
		location: [lat, lon]
	};
}

// Then to use it on the map page:
mapMarkers = data.mapMarkers.map(convertObjectToMarker);
```

Reply:
I personally prefer having everything in a contained class instead of separate type definitions and methods tbh. Classes also provide more flexibility in case more attributes get added to the array in the future.

Reply:
Hmm I mainly dislike the `from` and `fromMany` methods, because they're pretty unusual. Besides, this code isn't directly related to the API and may as well be in it's own file. And maybe remove decorator part from the name, since it's not a decorator.

In my opinion the following approach is more idiomatic, but all good if you keep it as is:

```typescript
export class MapMarker {
	private lat: number;
	private lon: number;
	public readonly assetId?: string;

	constructor(obj: object) {
		const [lat, lon, assetId] = obj as MapMarkerResponseDto;

		this.lat = lat;
		this.lon = lon;
		this.assetId = assetId;
	}

	// Can be accessed with marker.location to make destructuring easier
	get location(): [number, number] {
		return [this.lat, this.lon];
	}
}

// markers.map(obj => new MapMarker(obj))
```

Reply:
I agree that the term `decorator` is misleading here, therefore I renamed the class.